src/bootstrap/src/core/builder.rs

```diff
@@ -288,16 +288,61 @@ impl PathSet {
     }
 }
 
-const PATH_REMAP: &[(&str, &str)] = &[("rust-analyzer-proc-macro-srv", "proc-macro-srv-cli")];
-
-fn remap_paths(paths: &mut [&Path]) {
-    for path in paths.iter_mut() {
+const PATH_REMAP: &[(&str, &[&str])] = &[
+    // config.toml uses `rust-analyzer-proc-macro-srv`, but the
+    // actual path is `proc-macro-srv-cli`
+    ("rust-analyzer-proc-macro-srv", &["proc-macro-srv-cli"]),
+    // Make `x test tests` function the same as `x t tests/*`
+    (
+        "tests",
+        &[
+            "tests/assembly",
+            "tests/codegen",
```

Comment:
Nothing can go wrong with hardcoded paths.

Reply:
Maybe collects folders in /tests at runtime and assert that lists equal (excluding `auxiliary`)?

Reply:
We have a test for checking the existance paths. Seems like we need to test for missing paths too.

Reply:
> We have a test for checking the existance paths. Seems like we need a test for missing paths too.

Here it is: https://github.com/rust-lang/rust/pull/124969